DS-Session4-Question2/src/meta/utility/UniqueChar.java

```diff
@@ -0,0 +1,54 @@
+/**
+ * This Class Contains Methods to Calculate Unique Characters in a string 
+ * if string is searched before then it will returns you in O(1)
+ * else it will take O(n) Complexity 
+ * @author Abhishek Bishnoi
+ */
+package meta.utility;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class UniqueChar {
+	// cache map to store cache results
+	Map<String, Integer> cacheResult = new HashMap<String, Integer>();
+	/**
+	 * This Function Counts and return Unique characters in string 
+	 * @param string To check Unique Characters
+	 * @return unique character count
+	 */
+	public int countUniqueCharacters(String string){
+		try{
+			if(cacheResult.containsKey(string)){
+				return cacheResult.get(string);
+			}else{
+				int countUnique = 0;
+				for(int i = 0 ; i < string.length() ; i++){
+					if(search(string.charAt(i), string) == 1){
```

Comment:
Few points:
- You can convert string into character array.
- Create a map, where key of the map will be the characters of the array and value will be character count
- now iterate on the character array;
- if(map.contains(char[i])), then remove the character from the map
- else add the character in the map.
At the end size of map would be the size of unique characters.
Using this approach will save you from writing custom search function. And this is the requirement of the question.